gcore/gdalrasterband.cpp

```diff
@@ -2490,6 +2490,113 @@ uint64_t CPL_STDCALL GDALGetRasterNoDataValueAsUInt64(GDALRasterBandH hBand,
     return poBand->GetNoDataValueAsUInt64(pbSuccess);
 }
 
+/************************************************************************/
+/*                        SetNoDataValueAsString()                      */
+/************************************************************************/
+
+/**
+ * \brief Set the no data value for this band.
+ *
+ * Depending on drivers, changing the no data value may or may not have an
+ * effect on the pixel values of a raster that has just been created. It is
+ * thus advised to explicitly called Fill() if the intent is to initialize
+ * the raster to the nodata value.
+ * In any case, changing an existing no data value, when one already exists and
+ * the dataset exists or has been initialized, has no effect on the pixel whose
+ * value matched the previous nodata value.
+ *
+ * To clear the nodata value, use DeleteNoDataValue().
+ *
+ * @param pszNoData the value to set.
+ * @param[out] pbCannotBeExactlyRepresented Pointer to a boolean, or nullptr.
+ *             If the value cannot be exactly represented on the output data
+ *             type, *pbCannotBeExactlyRepresented will be set to true.
+ *
+ * @return CE_None on success, or CE_Failure on failure.  If unsupported
+ * by the driver, CE_Failure is returned but no error message will have
+ * been emitted.
+ *
+ * @since 3.11
+ */
+
+CPLErr
+GDALRasterBand::SetNoDataValueAsString(const char *pszNoData,
```

Comment:
how are errors handled/notified in case the given string is not convertible to the desired data type?

Reply:
> how are errors handled/notified in case the given string is not convertible to the desired data type?

through the pbCannotBeExactlyRepresented flag

Reply:
I was thinking at a string that cannot be converted to a number, not an out-of-bounds one.

Reply:
> I was thinking at a string that cannot be converted to a number,

*pbCannotBeExactlyRepresented should be set in that case too